pkg/apis/v1alpha1/managedzone_types.go

```diff
@@ -39,6 +39,15 @@ type ManagedZoneSpec struct {
 	// Reference to another managed zone that this managed zone belongs to.
 	// +optional
 	ParentManagedZone *ManagedZoneReference `json:"parentManagedZone,omitempty"`
+	// +required
+	SecretRef *SecretRef `json:"provider"`
```

Comment:
Why provider? Feels like it should be a bit more explicit that it's a reference to a secret

Reply:
no particular reason just a name i chose since its to do with the provider i can change it to make it clearer